src/libs/Navigation/NavigationRoot.tsx

```diff
@@ -76,26 +80,44 @@ function NavigationRoot({authenticated, lastVisitedPath, initialUrl, onReady}: N
     const currentReportIDValue = useCurrentReportID();
     const {isSmallScreenWidth} = useWindowDimensions();
     const {setActiveWorkspaceID} = useActiveWorkspace();
+    const [user] = useOnyx(ONYXKEYS.USER);
 
-    const initialState = useMemo(
-        () => {
-            if (!lastVisitedPath) {
-                return undefined;
-            }
+    const [hasCompletedGuidedSetupFlow] = useOnyx(ONYXKEYS.NVP_ONBOARDING, {
+        selector: hasCompletedGuidedSetupFlowSelector,
+    });
 
-            const path = initialUrl ? getPathFromURL(initialUrl) : null;
-
-            // For non-nullable paths we don't want to set initial state
-            if (path) {
-                return;
-            }
+    const initialState = useMemo(() => {
+        if (!user || user.isFromPublicDomain) {
+            return;
+        }
```

Comment:
This return is too early and prevented public domain users from using the `lastVisitedPath` feature.

**Repro steps:**
1. Go to settings
2. Open https://dev.new.expensify.com:8082/
3. Verify that you land on settings

This only works with private domain users. cc @mountiny 


https://github.com/user-attachments/assets/fd4e6d69-4a42-492e-bed3-e88144e6974c